modules/ROOT/pages/dataweave-extension-plugin.adoc

```diff
@@ -0,0 +1,400 @@
+
+[[scenarios]]
+== Understanding DataWeave Scenarios
+
+A DataWeave _scenario_ is a group of inputs and outputs that you can inject into your mapping either to try out in the *Preview Output* view or to test your mapping using the input to evaluate it and compare its output with the desired saved output of your scenario. Scenarios also provide autocompletion based on the input structures.
+
+These scenarios and their resources live in the `src/test/resources` folder and have the following structure:
+
+```
+├── src/test/resources
+│       └── MyMapping //Contains the different scenarios for your mapping called "MyMapping"
+│           ├── default //Name of your scenario
+│           │   └── inputs //Folder where all of your inputs live
+│           │   │   └── payload.json //Input for your mapping (you can use it in your script, naming it to be the same as the name of the file without the file's extension)
+│           │   │   └── vars //Folder that represents a name for your input variable in your script: in this case vars
+│           │   │       └── test.json //Input for your mapping (you can use it in your script, calling it vars.test)
+│           │   └── out.json // Expected output of your mapping running against your inputs (only present when testing)
+│           └── SavedTestScenario //Another scenario used for the integration test
+```
+=== Using the DataWeave Scenarios View
+
+The *DATAWEAVE SCENARIOS* view shows the available scenarios for the `.dwl` file. In this view, you can add, remove, or set the active scenario that is used for your preview execution, or you can choose that the values be autocompleted.
+
+image::dataweave-extension-dataweavescenario.png[DataWeave Scenarios view in Visual Studio Code]
+
+=== Using Reader Properties for Inputs
+
+If you want to set different reader properties to the inputs you use, create a properties file with the suffix `{fileName}-config.properties` in the same directory of your input. For example, if your input is called `payload.csv` then create a `payload-config.properties` file that contains all the reader properties required to run your scenario. Creating the properties file also applies while testing your DataWeave mappings.
+
+
+== Preview a DataWeave Mapping
+
+After you define sample data, get feedback on how your mapping is working by previewing the output.
+
+. In Visual Studio Code, open your DataWeave project.
+. Navigate to the `src/test/dw` folder.
+. Create a new preview `.dwl` file, for example `MyMappingPreview.dwl`. +
+ This new preview file imports the function you created in the `.dwl` file under the `src/main/dw` folder.
+. In the preview `.dwl` file, add your preview mapping code to try out the function.
+. Either preview the mapping function by one of these options:
+* Click the *DataWeave: Run Preview* button (arrow icon) at the top right corner of your preview mapping `.dwl` file.
+
```

Comment:
```suggestion
. Select your mapping `.dwl` file.
. Either preview the mapping function by one of these options:

```
Those steps were just for Previewing module's functions.
Removed duplicate step of "Click the *DataWeave: Run Preview*"